CMakeLists.txt

```diff
@@ -63,6 +63,27 @@ endif()
 
 if(UTF8PROC_ENABLE_TESTING)
   enable_testing()
+  if (NOT WIN32)
+    set(curl_cmd "/usr/bin/curl")
+    set(curl_param --retry 5 --location -\#)
+    set(UNICODE_VERSION 12.1.0)
+    set(UNICODE_EMOJI_VERSION 12.0)
+
+    execute_process(COMMAND mkdir -p data)
+    execute_process(
+      COMMAND ${curl_cmd} ${curl_param} -o data/emoji-data.txt -O http://unicode.org/Public/emoji/${UNICODE_EMOJI_VERSION}/emoji-data.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/UnicodeData.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/UnicodeData.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/EastAsianWidth.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/EastAsianWidth.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/GraphemeBreakProperty.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/auxiliary/GraphemeBreakProperty.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/DerivedCoreProperties.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/DerivedCoreProperties.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/CompositionExclusions.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/CompositionExclusions.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/CaseFolding.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/CaseFolding.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/NormalizationTest.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/NormalizationTest.txt
+      COMMAND ${curl_cmd} ${curl_param} -o data/GraphemeBreakTestOrg.txt -O http://www.unicode.org/Public/${UNICODE_VERSION}/ucd/auxiliary/GraphemeBreakTest.txt
```

Comment:
You don't need to download all of these.  Only `NormalizationTest.txt` and `GraphemeBreakTest.txt` are needed for the tests.

(The other files are only needed for re-generating the Unicode data tables, which will be unnecessary in `cmake` for the foreseeable future.)